src/Files.App/Helpers/FileSecurityHelpers.cs

```diff
@@ -60,166 +36,227 @@ public static string GetOwner(string path)
+				if (header.AceFlags.HasFlag(SystemSecurity.AceFlags.ObjectInherit))
+					inheritanceFlags |= AccessControlEntryFlags.ObjectInherit;
+				if (header.AceFlags.HasFlag(SystemSecurity.AceFlags.NoPropagateInherit))
+					inheritanceFlags |= AccessControlEntryFlags.NoPropagateInherit;
+				if (header.AceFlags.HasFlag(SystemSecurity.AceFlags.InheritOnly))
+					inheritanceFlags |= AccessControlEntryFlags.InheritOnly;
 
 				// Initialize an ACE
-				acl.AccessControlEntries.Add(new(isFolder, szSid, type, accessMaskFlags, isInherited, inheritanceFlags, propagationFlags));
+				aces.Add(new(isFolder, szSid, type, accessMaskFlags, isInherited, inheritanceFlags));
 			}
 
-			return acl;
-		}
+			// Initialize with proper data
+			acl = new AccessControlList(path, isFolder, principal, isValidAcl);
 
-		public static bool SetAccessControlList(AccessControlList acl)
-		{
-			return false;
+			// Set access control entries
+			foreach (var ace in aces)
+				acl.AccessControlEntries.Add(ace);
+
+			return Kernel32.GetLastError();
 		}
 
+		/// <summary>
+		/// Get access control list (ACL) initialized with default data.
+		/// </summary>
+		/// <param name="isFolder"></param>
+		/// <param name="ownerSid"></param>
+		/// <returns>If the function succeeds, an instance of AccessControlList; otherwise, null.</returns>
 		public static AccessControlEntry InitializeDefaultAccessControlEntry(bool isFolder, string ownerSid)
 		{
 			return new(
 				isFolder,
 				ownerSid,
-				FilesSecurity.AccessControlType.Allow,
-				FilesSecurity.AccessMaskFlags.ReadAndExecute,
+				AccessControlEntryType.Allow,
+				AccessMaskFlags.ReadAndExecute,
 				false,
 				isFolder
-					? FilesSecurity.InheritanceFlags.ContainerInherit | FilesSecurity.InheritanceFlags.ObjectInherit
-					: FilesSecurity.InheritanceFlags.None,
-				FilesSecurity.PropagationFlags.None);
+					? AccessControlEntryFlags.ContainerInherit | AccessControlEntryFlags.ObjectInherit
+					: AccessControlEntryFlags.None);
 		}
 
-		public static bool AddAccessControlEntry(string path, AccessControlEntry entry)
+		/// <summary>
+		/// Add an default Access Control Entry (ACE) to the specified object's DACL
+		/// </summary>
+		/// <param name="path">The object's path to add an new ACE to its DACL</param>
+		/// <param name="sid">Principal's SID</param>
+		/// <returns> If the function succeeds, the return value is ERROR_SUCCESS. If the function fails, the return value is a nonzero error code defined in WinError.h.</returns>
+		public static Win32Error AddAccessControlEntry(string szPath, string szSid)
 		{
-			// Get DACL
-			GetNamedSecurityInfo(
-				path,
+			// Get DACL for the specified object
+			var result = GetNamedSecurityInfo(
+				szPath,
 				SE_OBJECT_TYPE.SE_FILE_OBJECT,
 				SECURITY_INFORMATION.DACL_SECURITY_INFORMATION | SECURITY_INFORMATION.PROTECTED_DACL_SECURITY_INFORMATION,
 				out _,
 				out _,
-				out var pDacl,
+				out PACL pDACL,
 				out _,
 				out _);
```

Comment:
Could be one lined.

Reply:
This is convention when calling Win32API.